### PR TITLE
Resolve #450 #451 -- Disarm fix and Silence CC coded

### LINKED ISSUE #450: Fix Disarm

### PR DESCRIPTION
**Pull request by Deudly**
Disarm fix and Silence CC added

Refs #450 #451

### CODE REVIEW

File: GameServerLib/Logic/GameObjects/Champion.cs [hunk 1]
```diff
@@ -535,6 +535,10 @@ public override void onCollision(GameObject collider)
 
         public override void DealDamageTo(Unit target, float damage, DamageType type, DamageSource source, bool isCrit)
         {
+            if (HasCrowdControl(CrowdControlType.Disarm))
```
Comment (Heymeelie) [inline GameServerLib/Logic/GameObjects/Champion.cs:533]: If I'm not wrong, disarm is only with AA which means you can deal damages with abilities

**Review by Heymeelie — CHANGES_REQUESTED** [review]

File: GameServerLib/Logic/GameObjects/Champion.cs (hunk 1 above)
Comment (Deudly) [reply GameServerLib/Logic/GameObjects/Champion.cs:533]: Please, this event is called only by AA. Spells dont call this event @Neekhaulas

Comment (molenzwiebel) [reply GameServerLib/Logic/GameObjects/Champion.cs:533]: If spells don't call this event, it should be named differently.

Comment (Deudly) [reply GameServerLib/Logic/GameObjects/Champion.cs:533]: @molenzwiebel But this PR is not about solving it

Comment (moonshadow565) [reply GameServerLib/Logic/GameObjects/Champion.cs:533]: AAs are spells

Comment (Deudly) [reply GameServerLib/Logic/GameObjects/Champion.cs:533]: nvm, this shit works @Neekhaulas

**Pull request closed without merging**